api-gateway/src/main/resources/static/scripts/visits/visits.controller.js

```diff
@@ -682,7 +730,73 @@ angular.module('visits')
             }, function () {
                 console.log("Failed to create corresponding bill!");
             });
-        }
+        }*/
+
+        self.someFunction = function() {
+            console.log('Using ownerId in someFunction:', self.ownerId);
+            console.log('Using petId in someFunction:', self.petId);
+        };
+        $scope.goBack = function() {
+            $state.go('visitList');
+        };
+
+        self.submit = function () {
+            // Ensure ownerId and petId are set correctly when owner and pet are selected
+            self.someFunction();
+
+
+            // Check if ownerId and petId are set before creating the URL and proceeding with the POST request
+            var postURL = "api/gateway/visit/owners/" + self.ownerId + "/pets/" + self.petId + "/visits";
+            var billsUrl = "api/gateway/bills"; // Replace with your actual URL for bill creation
+
+            var data = {
+                visitDate: $filter('date')(self.chosenDate, "yyyy-MM-dd HH:mm"),
+                description: self.desc,
+                petId: self.petId,
+                practitionerId: self.practitionerId,
+                status: 'UPCOMING'
+            };
+
+            var billData = {
+                ownerId: self.ownerId, // This should be the ownerId, as fetched from the selected owner
+                date: $filter('date')(self.chosenDate, "yyyy-MM-dd"),
+                visitType: $("#selectedVisitType").val() // Consider using AngularJS data binding instead of directly accessing the DOM with jQuery
+            }
+
+            // Proceed with your existing logic for the POST request and response handling
+            $http.post(postURL, data).then(function (response) {
+                console.log(response)
+                $state.go('visitList');
+/*                let currentDate = getCurrentDate();
+
+                // Add the visit to one of the lists depending on its date
```

Comment:
I see that you have commented out the event of when the post does succeed. This is what displayed the green message and also displayed the list of added Visits at the bottom. Was this due to it not functioning properly?

Reply:
Yes, even before commenting these lines out the successful message does not pop up, so instead i made it so that it routes back to the  visits list page if successful and if not the corresponding message will be displayed.